src/main/java/ui/components/PanelNameTextField.java

```diff
@@ -5,36 +5,50 @@
 import javafx.application.Platform;
 import javafx.scene.control.TextField;
 import javafx.scene.input.KeyCode;
+import javafx.scene.layout.HBox;
 
 public class PanelNameTextField extends TextField {
 
     private final FilterPanel panel;
 
-    public PanelNameTextField(String panelName, FilterPanel panel) {
+    public PanelNameTextField(String panelName, HBox undoButton, HBox confirmButton, FilterPanel panel) {
```

Comment:
These aren't conceptually part of the text field, so they shouldn't be passed in here.

setNewName should be a method of FilterPanel which queries the text field for its value. That way this class is just a thin wrapper for a vanilla text field which handles key events and calls FilterPanel methods.

Querying for the text field value can be done via a new method I guess, since trim needs to be called. The method can return Optional<String>.

The button events can be bound in FilterPanel.